lib/travis/build/appliances/disable_apt_sources.rb

```diff
@@ -0,0 +1,27 @@
+require 'travis/build/appliances/base'
+
+module Travis
+  module Build
+    module Appliances
+      class DisableAptSources < Base
+        def apply
+          sh.raw <<~BASH
```

Comment:
I agree with the spirit of this PR, but I do have reservations about merging it right away.

This is going to break those builds that rely on these sources being available, so we should start with a warning, and give some time to adjust; e.g.,

    We will remove APT sources at YYYYMMDD. If your builds rely on them,
    we suggest adding them with `apt` addon. For more information, read DOC_LINK